benchmarking/src/result.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022-2023 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+//! Statistics from a single benchmark run, mostly gathered through `getrusage`.
+
+use std::{error::Error, time::Duration};
+
+use serde::{Deserialize, Serialize};
+
+/// `RunMeasurement` holds the statistics captured from a single run. This includes
+/// resource usage through `getrusage` (the same system call that powers
+/// `time`), as well as basic info about whether the process exited with an
+/// error or not.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RunMeasurement {
+    /// Elapsed time of run.
+    pub real_time: Duration,
+    /// User time (including all child processes like solvers).
+    pub user_time: Duration,
+    /// System time (including all child processes like solvers).
+    pub sys_time: Duration,
+    /// User time (excluding child processes).
+    pub self_user_time: Duration,
+    /// Max memory usage (RSS) in bytes
+    pub max_mem_bytes: usize,
+    /// True if the run was killed early due to a timeout.
+    pub timed_out: bool,
+    /// True if the run was successful
+    pub success: bool,
```

Comment:
Related to my other concern about timeout vs success in `time_bench.rs`, I'm wondering whether it's bad if both these booleans are true, and how the caller should interpret that?

I was able to induce this behavior in a few tries as follows:

```bash
$ cargo build -r --bin time 
$ ./target/release/time -t 1003ms --json sleep 1 | jq
could not send SIGTERM: EPERM: Operation not permitted
{
  ...
  "timed_out": true,
  "success": true
}
```
I had to play with the exact number of milliseconds to get this to work only some of the time and not all the time in one direction or the other.